Source/JavaScriptCore/API/JSContext.mm

```diff
@@ -343,23 +349,9 @@ @implementation JSContext (Internal)
 
 - (instancetype)initWithGlobalContextRef:(JSGlobalContextRef)context
 {
-    self = [super init];
-    if (!self)
-        return nil;
-
     JSC::JSGlobalObject* globalObject = toJS(context);
-    m_virtualMachine = [JSVirtualMachine virtualMachineWithContextGroupRef:toRef(&globalObject->vm())];
-    ASSERT(m_virtualMachine);
```

Comment:
This code is not the same, right? Why are we removing these assertions?

Reply:
Designated initializer nonsense. I just refactored the common logic into one method and called that the initializer.

Reply:
Fixed!